Add webnative auth

bgins opened the pull request:
## Summary

This PR implements the following features

* [x] Add webnative auth
* [x] Add page placeholders
* [x] Add FIL balance indicators to navbar

<!-- You can skip this if you're fixing a typo or adding an app to the Showcase. -->

This PR is mostly about adding webnative auth. A bit of work was also done scaffolding the UI.

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

## Test plan (required)

Loading (initializing webnative):

![loading](https://user-images.githubusercontent.com/7957636/106925721-33e26e80-66c5-11eb-9c0f-450bd22362d3.png)

Guest (has not signed in):
![guest](https://user-images.githubusercontent.com/7957636/106925435-f1b92d00-66c4-11eb-95ca-4e9e0a2c98ca.png)

Authed (has signed in):
![authed](https://user-images.githubusercontent.com/7957636/106925547-0ac1de00-66c5-11eb-81ed-b574dfc3e69a.png)

Placeholder pages for balances and files not shown. They are empty.

## After Merge


* [ ] Does this change invalidate any docs or tutorials? _If so ensure the changes needed are either made or recorded_
* [ ] Does this change require a release to be made? Is so please create and deploy the release

bgins committed:
Add process.browser types

We can check if we are in the browser or on the server by checking
process.browser in a conditional. This lets us work with browser only packages
like webnative.

bgins committed:
Add webnative auth

Add webnative auth module and update homepage with four possible states:
loading, authed, error, or otherwise guest. We show a spinner when loading,
a placeholder skeleton of cards when authed, an error message, or
a welcome message and sign in button to guests.

bgins committed:
Add placeholder balances and files pages

Add navbar links to the new pages.

bgins committed:
Add filecoin balance indicators

We want to treat balances as global information to make it easily available
when not on the balances page. These placeholders are one way to do it.

bgins committed:
Add balance indicator font color

walkah commented on src/routes/_layout.svelte:
```diff
@@ -1,29 +1,23 @@
 <script lang="ts">
   export let segment = undefined;
```
This line seems to be triggering warnings for me locally (running `npm run dev`):

> Header_1 has unused export property 'segment'. If it is for external reference only, please consider using `export const segment`

bgins replied:
Tracked down the warning. It looks like using `segment` in this way has been deprecated, and there is a `page` store that takes its place (https://github.com/sveltejs/sapper/issues/824). The `segment` and its replacement are used for getting access to  parts of the URL.

Will remove it and look into replacing it with the `page` store.

walkah reviewed (approved):
![tenor-241959124](https://user-images.githubusercontent.com/15350/106952497-2f3aac00-66ff-11eb-9529-948d9c1dede0.gif)

bgins committed:
Remove segment prop from layout and header

The use of segment has been deprecated in favor of the page store. We were not
using it in the header, but the idea is that it would be used for highlighting
the nav item for the active page.
